### PR TITLE
Add database plotting functions and improved ´consolidate()´ function

### LINKED ISSUE #134: Create a data_evolution() function that contrasts the processed (and exported) datasets to their original versions
- [ ] data_evolution()
  - This function will compare the evolution from the originally imported dataset to the final product we display in our packages to ensure transparency.
  - This will contain information such as the link to the original datasets, the original coding variables, etc.
  - Will make some changes to either the `export_data()` function to store the preparation script as an attribute for now or to the preparation script templates so that the initial variables are correctly saved as attributes to the exported object

### LINKED ISSUE #212: Update other workflow action files to be similar to prchecks.yml in regards to package caching

### LINKED ISSUE #214: Add visuals for databases/datasets
Visual functions should show what is already included in the data, to help us 
diagnose where further development is needed in the data (i.e. what is missing and where) and to communicate better why the package is important/impactful.

### LINKED ISSUE #215: Should be able to install develop versions from `get_packages()`

### LINKED ISSUE #218: `data_source()` and `data_contrast()` do not work with package caching in CI
Both functions rely on `find.package()` but since we are, often, looking for packages not on CRAN we cannot implement package caching for some of these packages. For now, on `{manytrade}` these code chunks were set not to run so that the readme file renders properly, but we should try to find a solution for this either in the function or in the CI.

### LINKED ISSUE #219: Improve printing for `get_packages()`
Make so that description for packages comes at the end and improve the line sizes and dividers.

### LINKED ISSUE #222: Update network_map() function
Function requires some packages not imported in manydata and, more importantly, it relies on a function in manystates to download cshape files... 

@jaeltan could you please make sure this is a feasible function to add to manydata considering CRAN requirements? If it is, we need to add the necessary package dependencies, update documentation, and helper functions.

### LINKED ISSUE #224: Setup `dbcomp()` to be consistent with ploting function as well as with new dataset structure
The code in the function not only needs to be made more concise but also, since `dbplot()` and `dbcomp()` are very similar, `dbcomp()` should be adapted so that it is called within `dbplot()` to avoid unnecessary code duplication across functions.

As well, with the new HUGGO and AIGOO structure, make sure the function does not return unnecessary hand-coded (already checked/worked on) obs in HUGGO to facilitate work on where it matters.

### LINKED ISSUE #225: Add HUGGO and AIGGO datasets across data packages
As a first step, we need to remove the current GNEVAR "consolidated" datasets across.

All hand-coded obs/variables should go into HUGGO.

All programmatically/automated code obs/variables should go on AIGGO

Concretely, for now, this means we move the accession and termination variables to AIGGO and the texts to HUGGO.

### LINKED ISSUE #227: Set a text argument in consolidate()
Now that we will have texts in HUGGO dataset, we should perhaps add an extra text = FALSE/TRUE argument to function so that users can decide if text variables should be used in consolidation. This should be false by default too make the process faster.

### LINKED ISSUE #228: Make `coalesce_compatible()` faster
For now, some consolidations in manyenviron take over 30 minutes and this is related to how we identify and coalesce duplicate rows. 

Using duplicated (with from last) is rather slow (see https://stackoverflow.com/questions/37148567/fastest-way-to-remove-all-duplicates-in-r) so perhaps we can speed things up using rle and sort (see https://stackoverflow.com/questions/1923273/counting-the-number-of-elements-with-the-values-of-x-in-a-vector).

### LINKED ISSUE #230: Is there a memory issue with ´consolidate()´?
In adding manyenviron HUGGO data, which is very long and wide, consolidate stops working... The errors appear to be related to memory limitations...

### LINKED ISSUE #240: Update cheat sheet
There are several changes and additions to manydata with the latest open PR. @jaeltan could you please update the cheat sheet?

### PR DESCRIPTION
**Pull request by henriquesposito**
# Description

These minor changes add functions to profile databases as plots or as a tibble, as well as improvements to how ´consolidate()´ works to make it faster and more capable of handling larger databases.

## Package

* Closed #212 by implementing package caching in GitHub actions workflows
* Closed #218 by fixing bug with GitHub actions workflows
* Closed #225 by changing the structure of datasets in "many" data packages
* Closed #240 by updating the package cheatsheet

## Connection

* Closed #134 by adding a ´data_evolution()´ function to the report family of functions that gets original datasets, if available, or opens the preparation scripts, if not available
* Added 'db_profile' family of functions to visualise databases
  * Closed #214 by adding ´db_plot()´ function to plot a profile of the database to facilitate comparison of matched observations across datasets
  * Closed #224 by adding ´db_comp()´ function that creates a tibble of the database to facilitate comparison of matched observations across datasets
* Updated ´get_packages()´ function
  * Closed #215 by making ´get_packages()´ interactive so that users can chose which branch to download
  * Closed #219 by improving ´get_packages()´ printing 
  * Updated ´get_packages()´ and ´plot_releases()´ to use ´{messydates}´, instead of ´{lubridate}´, for dates coercion
* Closed #222 by adding ´network_map()´function for plotting geographical networks
* Updated ´consolidate()´function to make function over 20 times faster
  * Closed #227 by making ´consolidate()´ ignore text related variables due to their size
  * Closed #230 by making ´consolidate()´ more concise to avoid running into memory limits
  * Closed #228 and #232 by replacing ´coalesce_compatible()´ for a faster approach to coalescing compatible missing observations that relies on ´zoo::na.locf()´
  * Made ´coalesce_compatible()´ function defunct

# Checklist:

- [x] PR form
  - [x] I have given this pull request an informative title
  - [x] Description above itemizes changes under subtitles, e.g. "## Collection""
  - [x] Any closed, fixed, or related issues are referenced and explained in the description above, e.g. "Fixed #0 by adding A"
  - [x] Package builds on my OS without issues
- [ ] PR checks all pass for latest commit
  - [x] Package builds on Mac
  - [x] Package builds on Windows
  - [x] Package builds on Linux
  - [ ] CodeCov check: Package improves or maintains good test coverage
  - [x] CodeFactor check: Package improves or maintains good style
- [x] Documentation
  - [x] Any new or modified functions or data have roxygen style documentation in their .R scripts
  - [x] Any longer functions are commented inline so that it is easier to debug in the future
  - [x] PR description above and the NEWS.md file are aligned
  - [x] DESCRIPTION file version is bumped by the appropriate increment (major, minor, patch)

### CODE REVIEW

File: data-raw/emperors/wikipedia/prepare-wikipedia.R [hunk 1]
```diff
@@ -23,15 +18,14 @@ wikipedia$notes
 # Some of the first few dates are supposed to be negative (BC).
 # While for other dates, only the year is accurate...
 # However, due to issues in dealing with dates with most R packages,
-# these dates became positive or were completed arbitrarly.
+# these dates became positive or were completed arbitrary.
```
Comment (jaeltan) [inline data-raw/emperors/wikipedia/prepare-wikipedia.R:21]: arbitrarily?

**Review by jaeltan — COMMENTED** [review]
Thank you @henriquesposito for all the time and hard work you've put into improving 'consolidate()' and 'db_plot()'/ 'db_comp()'! I made some very minor documentation comments that I can fix. 'consolidate()' also works fine with manytrade databases.

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+* Closed #212 by implementing package caching in GitHub actions workflows
+* Closed #218 by fixing bug with GitHub actions workflows
+* Closed #225 by changing the structure of datasets in "many" data packages
+
+## Connection
+
+* Closed #134 by adding a ´data_evolution()´ function to the report family of functions that gets original datasets, if available, or opens the preparation scripts, if not available
+* Added 'db_profile' family of functions to visualise databases
+  * Closed #214 by adding ´db_plot()´ function to plot a profile of the database
+  * Closed #224 by adding ´db_comp()´ function that creates a tibble comparing datasets
```
Comment (jhollway) [inline NEWS.md:14]: Explain when these would be used?

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+* Updated ´get_packages()´ function
+  * Closed #215 by making ´get_packages()´ interactive so that users can chose which branch to download
+  * Closed #219 by improving ´get_packages()´ printing
```
Comment (jhollway) [inline NEWS.md:17]: Can users choose several packages to download at once? E.g. `c(1,2,4)`?

File: NEWS.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+  * Closed #224 by adding ´db_comp()´ function that creates a tibble comparing datasets
+* Updated ´get_packages()´ function
+  * Closed #215 by making ´get_packages()´ interactive so that users can chose which branch to download
+  * Closed #219 by improving ´get_packages()´ printing 
+* Closed #222 by adding ´network_map()´function for plotting geographical networks
+* Updated ´consolidate()´function
+  * Closed #227 by making ´consolidate()´ ignore text related variables
+  * Closed #230 by re-working how ´consolidate()´ works to avoid running into memory limits
+  * Closed #228 and #232 by replacing ´coalesce_compatible()´ for a simpler and faster approach to coalescing compatible missing observations row wise
+  * Made ´coalesce_compatible()´ defunct
```
Comment (jhollway) [inline NEWS.md:23]: Could do with some more description/justification? Say how much faster... Is there any loss of functionality?

File: NEWS.md (hunk 2 above)
Comment (jhollway) [reply NEWS.md:14]: Can we add examples to README perhaps, or would this add an extra unnecessary dependency?

File: pkgdown/_pkgdown.yml [hunk 5]
```diff
@@ -64,17 +64,18 @@ reference:
     contents:
       - consolidate
       - pluck
-      - coalesce_compatible
       - favour
+      - starts_with("db_")
+      - starts_with("retrieve_")
```
Comment (jhollway) [inline pkgdown/_pkgdown.yml:69]: Aren't the `retrieve_*()` functions more about data wrangling?

File: pkgdown/_pkgdown.yml [hunk 6]
```diff
@@ -64,17 +64,18 @@ reference:
     contents:
       - consolidate
       - pluck
-      - coalesce_compatible
       - favour
+      - starts_with("db_")
```
Comment (jhollway) [inline pkgdown/_pkgdown.yml:68]: These functions are almost belonging to a new category?

File: NEWS.md (hunk 2 above)
Comment (henriquesposito) [reply NEWS.md:14]: Good idea! This was added to the readme.

**Review by jhollway — APPROVED** [review]

**Pull request merged**